rootfs/scheduler/resources/horizontalpodautoscaler.py

```diff
@@ -0,0 +1,154 @@
+            'spec': {
+                'minReplicas': min_replicas,
+                'maxReplicas': max_replicas,
+                'scaleRef': {
+                    # only works with Deployments, RS and RC
+                    'kind': target['kind'],
+                    'name': target['metadata']['name'],
+                    # the resource of the above which does the scale action
+                    'subresource': 'scale',
+                },
+                'cpuUtilization': {
+                    'targetPercentage': cpu_percent
+                }
+            }
+        }
+
+        return manifest
+
+    def create(self, namespace, name, target, **kwargs):
+        manifest = self.manifest(namespace, name, target, **kwargs)
+
+        url = self.api("/namespaces/{}/horizontalpodautoscalers", namespace)
+        response = self.session.post(url, json=manifest)
+        if self.unhealthy(response.status_code):
+            raise KubeHTTPException(
+                response,
+                'create HorizontalPodAutoscaler "{}" in Namespace "{}"', name, namespace
+            )
+            self.log(namespace, 'template used: {}'.format(json.dumps(manifest, indent=4)), 'DEBUG')  # noqa
+
+        # optionally wait for HPA if requested
+        if kwargs.get('wait', False):
+            self.wait(namespace, name)
+
+        return response
+
+    def update(self, namespace, name, target, **kwargs):
+        manifest = self.manifest(namespace, name, target, **kwargs)
+
+        url = self.api("/namespaces/{}/horizontalpodautoscalers/{}", namespace, name)
+        response = self.session.put(url, json=manifest)
+        if self.unhealthy(response.status_code):
+            self.log(namespace, 'template used: {}'.format(json.dumps(manifest, indent=4)), 'DEBUG')  # noqa
+            raise KubeHTTPException(response, 'update HorizontalPodAutoscaler "{}"', name)
+
+        # optionally wait for HPA if requested
+        if kwargs.get('wait', False):
+            self.wait(namespace, name)
+
+        return response
+
+    def delete(self, namespace, name):
+        url = self.api("/namespaces/{}/horizontalpodautoscalers/{}", namespace, name)
+        response = self.session.delete(url)
+        if self.unhealthy(response.status_code):
+            raise KubeHTTPException(
+                response,
+                'delete HorizontalPodAutoscaler "{}" in Namespace "{}"', name, namespace
+            )
+
+        return response
+
+    def wait(self, namespace, name):
+        # fetch HPA details
+        hpa = self.hpa.get(namespace, name).json()
+
+        # FIXME all of the below can be replaced with hpa['status'][desiredReplicas']
+        # when https://github.com/kubernetes/kubernetes/issues/29739 is fixed
+        # until then we have to query things ourselves
+
+        # only wait 30 seconds / attempts - this is not optimal
+        # ideally it would use the resources wait commands but they vary
+        for _ in range(30):
```

Comment:
`30` (and some other hard-coded numbers) keeps showing up here. Maybe it should be a module-level variable called `ATTEMPTS` or `RETRIES`?